src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/transaction_name_link.tsx

```diff
@@ -10,11 +10,14 @@
 import React from 'react';
 import { EuiLink, EuiText } from '@elastic/eui';
 import { getRouterLinkProps } from '@kbn/router-utils';
-import { TRANSACTION_DETAILS_BY_NAME_LOCATOR } from '@kbn/deeplinks-observability';
+import {
+  TRANSACTION_DETAILS_BY_NAME_LOCATOR,
+  type TransactionDetailsByNameParams,
+} from '@kbn/deeplinks-observability';
 import { getUnifiedDocViewerServices } from '../../../../plugin';
 
 interface TransactionNameLinkProps {
-  serviceName: string;
+  serviceName?: string;
```

Comment:
q: When will `serviceName` be undefined?

Reply:
When we are dealing with ad-hoc documents, such as those returned from ES|QL queries with `KEEP` filters applied (so you only get a subset of available fields from a document)